gm4_enchantment_extractors/data/gm4_enchantment_extractors/functions/extract_item.mcfunction

```diff
@@ -20,17 +24,18 @@ particle enchant ~ ~1.3 ~ 0 0 0 .5 10
 # failed extraction
 execute unless score $added_books gm4_ench_data matches 1.. run function gm4_enchantment_extractors:extract_failed
 
-# grant mending advancement
-data modify storage gm4_enchantment_extractors:temp block_items set from block ~ ~ ~ Items
-execute if data storage gm4_enchantment_extractors:temp block_items[{tag:{Enchantments:[{id:"minecraft:mending"}]}}] unless data storage gm4_enchantment_extractors:temp block_items[{tag:{Enchantments:[{id:"minecraft:mending"},{id:"minecraft:vanishing_curse"}]}}] run advancement grant @a[distance=..5] only gm4:enchantment_extractors_mending
-
 # update block inventory
 scoreboard players operation $slot_count gm4_ench_data += $added_books gm4_ench_data
 data remove storage gm4_enchantment_extractors:temp current_item.tag.Enchantments
 data modify block ~ ~ ~ Items append from storage gm4_enchantment_extractors:temp current_item
 
+# grant mending advancement
+data modify storage gm4_enchantment_extractors:temp block_items set from block ~ ~ ~ Items
+execute unless score $existing_mending gm4_ench_data matches 1.. if data storage gm4_enchantment_extractors:temp block_items[{tag:{StoredEnchantments:[{id:"minecraft:mending"}]}}] unless data storage gm4_enchantment_extractors:temp block_items[{tag:{StoredEnchantments:[{id:"minecraft:mending"},{id:"minecraft:vanishing_curse"}]}}] run advancement grant @a[distance=..5] only gm4:enchantment_extractors_mending
```

Comment:
Is it necessary to check for not `[{id:"minecraft:mending"},{id:"minecraft:vanishing_curse"}]` here? Just not `[{id:"minecraft:vanishing_curse"}]` should be sufficient.